_posts/2022-11-27-Diagrams-as-Code.md

```diff
@@ -0,0 +1,99 @@
+---
+title: "Tip of the Week: Diagrams as Code"
+author: Dave Bunten
+member: dave-bunten
+tags:
+  - tip-of-the-week
+  - software
+  - diagrams
+  - mermaid
+  - markdown
+---
+
+# Tip of the Week: Diagrams as Code
+
+> Each week we seek to provide a software tip of the week geared towards helping you achieve your software goals. If you have any software questions or suggestions for an upcoming tip of the week, please don’t hesitate to reach out to #software-engineering on Slack or email DBMISoftwareEngineering at olucdenver.onmicrosoft.com
+
+Diagrams can be a useful way to illuminate and communicate ideas. Free-form drawing or drag and drop tools are one common way to create diagrams. With this tip of the week we introduce another option: diagrams as code (DaC), or creating diagrams by using code.
+
+__TLDR (too long, didn't read);__
+Diagrams as code (DaC) tools provide an advantage for illustrating concepts by enabling __quick visual positioning__, __source controllable input__, __portability (both for input and output formats)__, and __open collaboration through reproducibility__. Consider using [Mermaid](https://mermaid-js.github.io/mermaid/) (as well as many other DaC tools) to assist your diagramming efforts which can be used directly, within in your markdown files, or [Github commentary](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams#creating-mermaid-diagrams) using code blocks (for example, ` ```mermaid `).
+
+## Example Mermaid Diagram as Code
+
+{% capture col1content %}
+
+```
+flowchart LR
+    a --> b
+    b --> c
+    c --> d1
+    c --> d2
+```
+
+_Mermaid code_
+{% endcapture %}
+{% capture col2content %}
+
+<pre class="mermaid">
+flowchart LR
+    a --> b
+    b --> c
+    c --> d1
+    c --> d2
+</pre>
+_Mermaid rendered_
+<script type="module">
+  import mermaid from 'https://unpkg.com/mermaid@9/dist/mermaid.esm.min.mjs';
+  mermaid.initialize({ startOnLoad: true });
+</script>
+
+{% endcapture %}
+{% include two-col.html col1=col1content col2=col2content %}
+
+The above shows an example [mermaid flowchart](https://mermaid-js.github.io/mermaid/#/flowchart) code and it's rendered output. The syntax is specific to mermaid and acts as a simple coding language to help you depict ideas. Mermaid also includes options for sequence, class, gantt, and other diagram types. Mermaid provides a [live editor](https://mermaid.live/edit#pako:eNpVzD1PwzAQBuC_Et2cRv6KL_HABBtT2SovV9vQCBxXlqPSRvnvhFRI9KZ73vuYwSUfwMD7V7q4E-VSve7tWK1F1W73VB3vOG5wd7gNnj9IQA0x5EiDX9_NvzML5RRisGDW1lP-tGDHZd2jqaS36-jAlDyFGqazpxKeB_rIFB_DFz-UlP-yM42HlOI_gpnhGww2vMMee62kapliqoYrmJ41SjONrOVSSN21Sw237Z43DLVgXCNKIWSHuPwAvLtMeg) which can be used to quickly draft and share content.
```

Comment:
Just a few minor things: "Gantt" is a proper noun so IMHO it should be capitalized. Also, I changed "it's" to the possessive form "its" rather than the contraction of "it is".
```suggestion
The above shows an example [mermaid flowchart](https://mermaid-js.github.io/mermaid/#/flowchart) code and its rendered output. The syntax is specific to mermaid and acts as a simple coding language to help you depict ideas. Mermaid also includes options for sequence, class, Gantt, and other diagram types. Mermaid provides a [live editor](https://mermaid.live/edit#pako:eNpVzD1PwzAQBuC_Et2cRv6KL_HABBtT2SovV9vQCBxXlqPSRvnvhFRI9KZ73vuYwSUfwMD7V7q4E-VSve7tWK1F1W73VB3vOG5wd7gNnj9IQA0x5EiDX9_NvzML5RRisGDW1lP-tGDHZd2jqaS36-jAlDyFGqazpxKeB_rIFB_DFz-UlP-yM42HlOI_gpnhGww2vMMee62kapliqoYrmJ41SjONrOVSSN21Sw237Z43DLVgXCNKIWSHuPwAvLtMeg) which can be used to quickly draft and share content.
```

Reply:
Thank you for catching these! I'll add this into a forthcoming commit.